api/management/commands/csv_ingest.py

```diff
@@ -1,21 +1,43 @@
 from django.core.management.base import BaseCommand
+from django.core import management
 import csv
+
 from api.models import Location, Hour
-import os
 
 
 class Command(BaseCommand):
-    help = 'Imports CSV file to populate local DB'
+    help = 'Ingests data into the db from csv_files. This command will kill and fill the db.'
 
     def handle(self, *args, **options):
-        for filename in ['community_gardens.csv']:
-            print('importing CSV file {0}'.format(filename))
-            with open(os.path.join('csv_files', filename), 'r') as f:
-                reader = csv.reader(f)
-                next(f)  # skips first line so we don't import headers
-                for r in reader:
-                    print('populating row: {0}'.format(r))
-                    loc = Location(name=r[1], category=r[2], address_1=r[3],
-                                   city=r[4], state=r[5], zipcode=r[6],
-                                   latitude=float(r[7]), longitude=float(r[8]))
-                    loc.save()
+        # this does a full kill and fill on the database.
+        management.call_command('flush', verbosity=0, interactive=False)
+        csv_files = ['farmers-market.csv',
+                     'community_gardens.csv', 'food-pantry.csv', 'supermarket.csv']
+        location_count = 0
+        for file in csv_files:
+            try:
+                with open('csv_files/{}'.format(file), 'r') as f:
+                    reader = csv.reader(f)
+                    next(f)
+                    for r in reader:
+                        loc = Location(name=r[0], address_1=r[1], address_2=r[2], city=r[3], state=r[
+                                       4], zipcode=r[5], phone=r[6], latitude=r[7], longitude=r[8], category=r[9], website=r[10], active=True)
+                        loc.save()
+                        hours = list(filter(None, r[11:]))
+                        for i in [hours[i:i + 3] for i in range(0, len(hours), 3)]:
```

Comment:
I find this nested for comprehension a bit hard to follow, especially since it's not clear what the 3 pieces of the Hour data are when we start slicing them out.

I wonder if something like this might work as a way of labeling more clearly what those pieces are

```python
                        for i in range(0, len(hours), 3):
                            hour_components = hours[i:i + 3]
                            (day, open_time, close_time) = hour_components
                            hour = Hour(day=day,
                                     open_time=self._format_time(open_time),
                                     close_time=self._format_time(close_time),
                                     location=loc)
                            hour.save()
```